Makefile

```diff
@@ -0,0 +1,163 @@
+# Are your paths correct?
+#
+# Makefile expects to run at the project root, and not nested inside anywhere
+#
+#  cd $DESK/xvm; make
+
+# Use bash for recipes
+SHELL := /bin/bash
+
+# Keep partial builds but not partial recipes
+.NOTINTERMEDIATE:	
+
+# for printing variable values
+# usage: make print-VARIABLE
+#        > VARIABLE = value_of_variable
+print-%  : ; @echo $* = $($*)
+
+# jar-file seperator - I can make this more specific for e.g. mac vs linux
+ifeq ($(OS),Windows_NT)
+SEP = ;
+else
+SEP = :
+endif
+
+# Handy for Cliff, $DESK/xvm
+# Forward slash instead of backslash for 
+DESK2 = $(subst \,/,$(DESK))
+
+# XVM project root
+XVM_ROOT = $(DESK2)/xvm
+
+# Find a reasonable ctags.
+CTAGS = $(shell which ctags)
+# Hack for MacOS: /usr/bin/ctags is unfriendly, so look for ctags from brew
+ifeq ($(UNAME),Darwin)
+	CTAGS = $(shell brew list ctags 2> /dev/null | grep bin/ctags)
+endif
+
+# Fun Args to javac.
+JAVAC_ARGS = -source 21 -target 21 -XDignore.symbol.file -Xlint:all -Xlint:-this-escape -Xlint:-unchecked -Xlint:-preview -Xlint:-deprecation -Xlint:-serial -Xlint:-rawtypes
+
```

Comment:
Why are you even turning on lint, if you remove a huge chunk of the more common warnings? As much as people expect code standards not being too weird in 2024 when they take a look at open source projects, they also tend to react negatively to a mixture of Java 1.4 and Java 21 code. We should strive to use generics where generics are suggested by the Java language, we should not use raw types, we should avoid escaping "this" from constructors, and we should design stuff so that having to do unchecked casts are a rare exception. We should certainly warn about deprecations, because it's bad if something stops working without warning when e.g. Java is upgraded.

If we do not want to follow Java programming practices, and it's motivated to turn off a lint check, then that should be done by @SuppressWarnings at the location of the code smell, so that it can be identified and refactored later. For example, I have some issues with "this-escape" because javax Inject makes some stateless/final things hard, but I have flagged it as a suppressedwarning in the code so that both things like SonarQube and future greps can identify the sites that should be given some refactoring love in the future.